testutil/base.go

```diff
@@ -60,3 +62,31 @@ func (s *BaseTest) TearDownTest(c *check.C) {
 func (s *BaseTest) AddCleanup(f func()) {
 	s.cleanupHandlers = append(s.cleanupHandlers, f)
 }
+
+// Mocks any mutable object with a compliant substitute, and insures it will be restored after the test
+func (s *BaseTest) MockAny(mockedByPtr interface{}, mock interface{}) {
```

Comment:
as explained elsewhere in best practice tests, things to mock wouldn't be accessible from the tests themselves but only from an export_test.go so I don't think we want this helper.